src/Microsoft.AspNet.Hosting/HostingEngine.cs

```diff
@@ -39,16 +40,16 @@ public IDisposable Start(HostingContext context)
             InitalizeServerFactory(context);
             EnsureApplicationDelegate(context);
 
-            var applicationLifetime = (ApplicationLifetime)context.Services.GetRequiredService<IApplicationLifetime>();
+            var applicationLifetime = (ApplicationLifetime)context.ApplicationLifeTime;
```

Comment:
Yeah. But the cast is not introduced with this change. So I would keep that as a separate work item rather than mixing it with API review.